apps/src/craft/code-connection/cc-client.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * CC client for 3rd party extension
+ *
+ */
+
+export default class cc_client {
+    constructor(port) {
+        this.port = port;
+        this.baseUrl = "http://localhost:";
+        this.asyncCallback = undefined;
+    }
+
+    /**
+     * Check the connection between Code Connection
+     *
+     * @param {callback} asyncCallback - callback to handle return value (true / false)
+     * @param {number} timeout - timeout for async ajax
+     *
+     */
+    connectionStatusUpdate(asyncCallback, timeout) {
+        this.asyncCallback = asyncCallback;
+        var that = this;
+        $.ajax({
+            type: "GET",
+            url: `${this.baseUrl}${this.port}/connected`,
+            timeout: timeout,
+            success: function (data) {
+                // cc responded
+                that.asyncCallback(data);
+                that.asyncCallback = undefined;
+            },
+            error: function (jqxhr, textStatus, error) {
+                // TODO: handle net::ERR_CONNECTION_REFUSED error gracefully
+                that.asyncCallback(false);
+                that.asyncCallback = undefined;
+            }
```

Comment:
I brought this code from our scratchx extension and there was some kind of scope issue so we added weird things like `that`, member `asyncCallback` and clean-up after each call. Let me double check one more time. I'm not using `connectionStatusUpdate` in this commit